packages-next/admin-ui/src/components/Pagination.tsx

```diff
@@ -0,0 +1,161 @@
+/** @jsx jsx */
+import { useEffect, useState } from 'react';
+import { jsx, Stack, useTheme } from '@keystone-ui/core';
+import { Select } from '@keystone-ui/fields';
+import { ChevronRightIcon, ChevronLeftIcon } from '@keystone-ui/icons';
+import { Link, useRouter } from '../router';
+
+interface PaginationProps {
+  pageSize: number;
+  total: number;
+  currentPage: number;
+  list: Record<string, any>;
+}
+
+const usePaginationStats = ({ list, pageSize, currentPage, total }: PaginationProps) => {
+  const [stats, setStats] = useState<string>('');
+  const { singular, plural } = list;
+  useEffect(() => {
+    if (total > pageSize) {
+      const start = pageSize * (currentPage - 1) + 1;
+      const end = Math.min(start + pageSize - 1, total);
+      setStats(`${start} - ${end} of ${total} ${list.plural}`);
+    } else {
+      if (total > 1 && list.plural) {
+        setStats(`${total} ${list.plural}`);
+      } else if (total === 1 && list.singular) {
+        setStats(`${total} ${list.singular}`);
+      }
+    }
+  }, [plural, singular, total, pageSize, currentPage]);
+
+  return { stats };
```

Comment:
Are you intentionally using state + an effect here for some reason or could these things be calculated in render and avoid the state?

Reply:
@mitchellhamilton mostly just to avoid calculating this multiple times if other nodes up the tree are liable to update.
Happy for this to occur in render and avoid state if that's preferred

Reply:
yeah, doing this in render would be preferred. (if this was actually expensive which I'm sure it's not then useMemo would be the right thing over state)

Reply:
done, function renamed as well so we don't confuse this with actual hook functionality